felt/src/lib.rs

```diff
@@ -231,91 +848,83 @@ mod test {
         fn bitand_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "(0|[1-9][0-9]*)"){
             let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
             let y = Felt::parse_bytes(y.as_bytes(), 10).unwrap();
-            let p = FeltBigInt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let result = &x & &y;
             result.to_biguint();
-            prop_assert!(result < p);
+            prop_assert!(result <= Felt::max_value());
         }
 
         #[test]
         // Property based test that ensures, for a pair of 100 values {x} and {y} generated at random each time tests are run, that performing a BitOr operation between them returns a result that is inside of the range [0, p].
         fn bitor_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "(0|[1-9][0-9]*)"){
             let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
             let y = Felt::parse_bytes(y.as_bytes(), 10).unwrap();
-            let p = FeltBigInt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let result = &x | &y;
             println!("x: {}, y: {}, result: {}", x, y, result);
             result.to_biguint();
-            prop_assert!(result < p);
+            prop_assert!(result <= Felt::max_value());
         }
 
         #[test]
         // Property based test that ensures, for a pair of 100 values {x} and {y} generated at random each time tests are run, that performing a BitXor operation between them returns a result that is inside of the range [0, p].
         fn bitxor_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "(0|[1-9][0-9]*)"){
             let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
             let y = Felt::parse_bytes(y.as_bytes(), 10).unwrap();
-            let p = FeltBigInt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let result = &x ^ &y;
             println!("x: {}, y: {}, result: {}", x, y, result);
             result.to_biguint();
-            prop_assert!(result < p);
+            prop_assert!(result <= Felt::max_value());
         }
 
         #[test]
          // Property-based test that ensures, for 100 values {x} that are randomly generated each time tests are run, that raising {x} to the {y}th power returns a result that is inside of the range [0, p].
         fn pow_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "[0-9]{1,2}"){
             let base = &Felt::parse_bytes(x.as_bytes(), 10).unwrap();
             let exponent:u32 = y.parse()?;
-            let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
 
             let result = Pow::pow(base, exponent);
             let as_uint = &result.to_biguint();
-            prop_assert!(as_uint < p, "{}", as_uint);
+            prop_assert!(as_uint <= &Felt::max_value().to_biguint(), "{}", as_uint);
```

Comment:
This was correct as it was. It compared the prime itself in `BigUint`, unmodified.

Reply:
Resolved: https://github.com/lambdaclass/cairo-rs/pull/743/commits/c268a2bb81cd3e1e23a667895ab2ac8228e8979e